cdap-api/src/main/java/co/cask/cdap/api/templates/AdapterConfigurer.java

```diff
@@ -112,7 +115,64 @@
    * @param datasetClass dataset class to create the Dataset type from
    * @param props dataset instance properties
    */
-  void createDataset(String datasetName,
-                     Class<? extends Dataset> datasetClass,
-                     DatasetProperties props);
+  void createDataset(String datasetName, Class<? extends Dataset> datasetClass, DatasetProperties props);
+
+  /**
+   * Adds a Plugin usage to the Adapter and create a new instance.
+   * The Plugin will be accessible at execution time via the {@link AdapterContext}.
+   *
+   * @param pluginType plugin type name
+   * @param pluginName plugin name
+   * @param properties properties for the plugin. The same set of properties will be used to instantiate the plugin
+   *                   instance at execution time
+   * @param <T> type of the plugin class
+   * @return A new instance of the plugin class or {@code null} if no plugin was found
+   */
+  @Nullable
+  <T> T usePlugin(String pluginType, String pluginName, PluginProperties properties);
```

Comment:
PipelineConfigurer is internal and specific for ETL app. The CDAP system doesn't know about it. It's the job of the ETL program to expose it through the `PipelineConfigurer` if needed.